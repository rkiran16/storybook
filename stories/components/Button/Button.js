import { html } from "lit-html";

export const Button = ({disabled = false, hasIcon = false, size = 'default', variation, label = "Primary Button", iconName = "add"}) => {
  const iconClasses = hasIcon ? "btn-w-icon d-flex justify-content-center align-items-center" : '';
  const sizeClass = size === 'skinny' && hasIcon ? 'btn-sm btn-w-icon-sm' : size === 'skinny' ? 'btn-sm' : '';

  return html`
    <button type="button" ?disabled=${disabled} class=${['btn', variation, iconClasses, sizeClass].join(' ')}>
      ${hasIcon ? html`<span class="material-icons md-16 me-2">${iconName}</span>` : ''}
      ${label}
    </button>
  `;
};
