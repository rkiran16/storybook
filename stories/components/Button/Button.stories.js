import { html } from 'lit-html';
import { Button } from  './Button';

export default {
  title: 'Components/Buttons/Button',
  parameters: {
    layout: 'centered',
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6297bdfb36de85127d3e27cc",
  },
  argTypes: {
    disabled: {
      type: { name: 'boolean', required: false},
      description: "Use when the user cannot proceed or until an input is collected",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
    },
    hasIcon: {
      name: "Leading Icon",
      type: { name: 'boolean', required: false},
      description: "Use when the user cannot proceed or until an input is collected",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
    },
    size: {
      name: "Size",
      defaultValue: 'default',
      control: { type: 'radio' },
      description: "Select Size",
      table: {
         type: { summary: 'radio' },
         defaultValue: { summary: 'default' },
      },
      options: ['default', 'skinny']
    },
    variation: {
       name: "Button Variations",
       defaultValue: "primary",
       control: { type: 'radio'},
       table: {
          type: { summary: 'radio' },
          defaultValue: { summary: 'primary' },
       },
       options: ['primary', 'secondary', 'tertiary'],
       mapping: {
          primary: 'btn-primary',
          secondary: 'btn-outline-secondary',
          tertiary: 'btn-tertiary'
       }
    }
  },
};

const Template = (args) => Button(args);


export const PrimaryButton = Template.bind({});
PrimaryButton.args = {}

export const SecondaryButton = Template.bind({});
SecondaryButton.args = {
   label: "Secondary Button",
   variation: "secondary"
}

export const TertiaryButton = Template.bind({});
TertiaryButton.args = {
   label: "Tertiary Button",
   variation: "tertiary"
}

const TertiaryButtonTemplate = args => {
   return html`
      <div class="d-flex w-100">
        <div class="me-3">${Button({label: "Upload button", iconName: 'file_upload', variation: "btn-tertiary", hasIcon: true})}</div>
        <div class="me-3">${Button({label: "Download button", iconName: 'file_download', variation: "btn-tertiary", hasIcon: true})}</div>
        <div class="me-3">${Button({label: "Show more", iconName: 'expand_more', variation: "btn-tertiary", hasIcon: true})}</div>
        <div class="me-3">${Button({label: "Show less", iconName: 'expand_less',  variation: "btn-tertiary", hasIcon: true})}</div>
      </div>
   `
}

export const TertiaryButtonVariations = TertiaryButtonTemplate.bind({});
TertiaryButtonVariations.args = {};