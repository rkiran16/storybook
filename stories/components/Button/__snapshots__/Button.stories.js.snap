// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Buttons/Button Primary Button 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <button type=\\"button\\" ?disabled=",
    " class=",
    ">
      ",
    "
      ",
    "
    </button>
  ",
  ],
  "values": Array [
    false,
    "btn btn-primary  ",
    "",
    "Primary Button",
  ],
}
`;

exports[`Storyshots Components/Buttons/Button Secondary Button 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <button type=\\"button\\" ?disabled=",
    " class=",
    ">
      ",
    "
      ",
    "
    </button>
  ",
  ],
  "values": Array [
    false,
    "btn btn-outline-secondary  ",
    "",
    "Secondary Button",
  ],
}
`;

exports[`Storyshots Components/Buttons/Button Tertiary Button 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <button type=\\"button\\" ?disabled=",
    " class=",
    ">
      ",
    "
      ",
    "
    </button>
  ",
  ],
  "values": Array [
    false,
    "btn btn-tertiary  ",
    "",
    "Tertiary Button",
  ],
}
`;

exports[`Storyshots Components/Buttons/Button Tertiary Button Variations 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <div class=\\"d-flex w-100\\">
        <div class=\\"me-3\\">",
    "</div>
        <div class=\\"me-3\\">",
    "</div>
        <div class=\\"me-3\\">",
    "</div>
        <div class=\\"me-3\\">",
    "</div>
      </div>
   ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <button type=\\"button\\" ?disabled=",
        " class=",
        ">
      ",
        "
      ",
        "
    </button>
  ",
      ],
      "values": Array [
        false,
        "btn btn-tertiary btn-w-icon d-flex justify-content-center align-items-center ",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<span class=\\"material-icons md-16 me-2\\">",
            "</span>",
          ],
          "values": Array [
            "file_upload",
          ],
        },
        "Upload button",
      ],
    },
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <button type=\\"button\\" ?disabled=",
        " class=",
        ">
      ",
        "
      ",
        "
    </button>
  ",
      ],
      "values": Array [
        false,
        "btn btn-tertiary btn-w-icon d-flex justify-content-center align-items-center ",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<span class=\\"material-icons md-16 me-2\\">",
            "</span>",
          ],
          "values": Array [
            "file_download",
          ],
        },
        "Download button",
      ],
    },
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <button type=\\"button\\" ?disabled=",
        " class=",
        ">
      ",
        "
      ",
        "
    </button>
  ",
      ],
      "values": Array [
        false,
        "btn btn-tertiary btn-w-icon d-flex justify-content-center align-items-center ",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<span class=\\"material-icons md-16 me-2\\">",
            "</span>",
          ],
          "values": Array [
            "expand_more",
          ],
        },
        "Show more",
      ],
    },
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <button type=\\"button\\" ?disabled=",
        " class=",
        ">
      ",
        "
      ",
        "
    </button>
  ",
      ],
      "values": Array [
        false,
        "btn btn-tertiary btn-w-icon d-flex justify-content-center align-items-center ",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<span class=\\"material-icons md-16 me-2\\">",
            "</span>",
          ],
          "values": Array [
            "expand_less",
          ],
        },
        "Show less",
      ],
    },
  ],
}
`;
